server/core/storage.go

```diff
@@ -485,10 +495,25 @@ func (s *Storage) SaveServiceGCSafePoint(ssp *ServiceSafePoint) error {
 
 // RemoveServiceGCSafePoint removes a GC safepoint for the service
 func (s *Storage) RemoveServiceGCSafePoint(serviceID string) error {
+	if serviceID == GCWorkerServiceSafePointID {
+		return errors.New("cannot remove service safe point of gc_worker")
+	}
 	key := path.Join(gcPath, "safe_point", "service", serviceID)
 	return s.Remove(key)
 }
 
+func (s *Storage) initServiceGCSafePointForGCWorker() (*ServiceSafePoint, error) {
+	ssp := &ServiceSafePoint{
+		ServiceID: GCWorkerServiceSafePointID,
+		SafePoint: 0,
+		ExpiredAt: math.MaxInt64,
+	}
+	if err := s.SaveServiceGCSafePoint(ssp); err != nil {
```

Comment:
return s.SaveServiceGCSafePoint(ssp)

Reply:
`SaveServiceGCSafePoint` doesn't return the ssp, and I think it's better to return nil for the `ssp` on error.

Reply:
oh, my mistake.